core/src/main/scala/com/salesforce/op/stages/impl/feature/TextTokenizer.scala

```diff
@@ -162,30 +162,84 @@ object TextTokenizer {
 
-        val sentences = sentenceSplitter.map(_.getSentences(lowerTxt, language))
-          .getOrElse(Seq(lowerTxt))
-          .map { sentence =>
-            val tokens = analyzer.analyze(sentence, language)
-            tokens.filter(_.length >= minTokenLength).toTextList
-          }
-        TextTokenizerResult(language, sentences)
-      case _ =>
-        TextTokenizerResult(defaultLanguage, Seq(TextList.empty))
+    val sentences = sentenceSplitter.map(_.getSentences(lowerTxt, language))
+      .getOrElse(Seq(lowerTxt))
+      .map { sentence =>
+        val tokens = analyzer.analyze(sentence, language)
+        tokens.filter(_.length >= minTokenLength).toTextList
+      }
+    TextTokenizerResult(language, sentences)
+  }
+
+  /**
+   * Language wise sentence tokenization
+   *
+   * @param textStringOpt       text to tokenize (in Option[String] form)
+   * @param languageDetector    language detector instance
+   * @param analyzer            text analyzer instance
+   * @param sentenceSplitter    sentence splitter instance
+   * @param autoDetectLanguage  whether to attempt language detection
+   * @param defaultLanguage     default language
+   * @param autoDetectThreshold language detection threshold
+   * @param toLowercase         whether to convert all characters to lowercase before tokenizing
+   * @param minTokenLength      minimum token length
+   * @return detected language and sentence tokens
+   */
+  def tokenizeStringOpt(
+    textStringOpt: Option[String],
+    languageDetector: LanguageDetector = LanguageDetector,
+    analyzer: TextAnalyzer = Analyzer,
+    sentenceSplitter: Option[SentenceSplitter] = None,
+    autoDetectLanguage: Boolean = AutoDetectLanguage,
+    defaultLanguage: Language = DefaultLanguage,
+    autoDetectThreshold: Double = AutoDetectThreshold,
+    toLowercase: Boolean = ToLowercase,
+    minTokenLength: Int = MinTokenLength
+  ): TextTokenizerResult = {
+    textStringOpt match {
+      case Some(txt) => tokenizeString(txt, languageDetector, analyzer, sentenceSplitter, autoDetectLanguage,
+        defaultLanguage, autoDetectThreshold, toLowercase, minTokenLength)
+      case None => TextTokenizerResult(defaultLanguage, Seq(TextList.empty))
     }
   }
 
+  /**
+   * Language wise sentence tokenization
+   *
+   * @param text                text to tokenize
+   * @param languageDetector    language detector instance
+   * @param analyzer            text analyzer instance
+   * @param sentenceSplitter    sentence splitter instance
+   * @param autoDetectLanguage  whether to attempt language detection
+   * @param defaultLanguage     default language
+   * @param autoDetectThreshold language detection threshold
+   * @param toLowercase         whether to convert all characters to lowercase before tokenizing
+   * @param minTokenLength      minimum token length
+   * @return detected language and sentence tokens
+   */
+  def tokenize(
```

Comment:
That's a bummer that you have to add so much boiler plate here. There are alternatives we can discuss.

Reply:
Yes, I was not too happy with this either, but could not think of a clean alternative.

Reply:
Before this change everything had to be wrapped into `Text` to call the `tokenize`. Then for `String` value would have require additional wrapping into `Option`, which was safer but less efficient. 
I cant find where do you use other overloading for `tokenize` with `String` and `Option[String]`. Is it in this PR?

Reply:
The `computeTextStats` and `computeTextMapStats` use them. I wanted to avoid having to wrap/unwrap values here since these functions are called per-entry.

Reply:
Makes sense then.